code/game/machinery/teleporter.dm

```diff
@@ -299,6 +322,13 @@
 		return
 	if (istype(M, /atom/movable))
 		if(do_teleport(M, com.target))
+			if(istype(M, /obj/structure/closet))
```

Comment:
I would suggest just looping trough the .contents of all atoms you can find recursivly, because you still have mechs and other things